python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+
+import uuid
+
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import validates
+from sqlalchemy.schema import FetchedValue
+from app.extensions import db
+
+from . import permit_amendment_status_code, permit_amendment_type_code
+from ....utils.models_mixins import AuditMixin, Base
+
+
+class PermitAmendment(AuditMixin, Base):
+    __tablename__ = 'permit_amendment'
+    permit_amendment_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    permit_amendment_guid = db.Column(UUID(as_uuid=True))
+    permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'), nullable=False)
+    received_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    issue_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    authorization_end_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    permit_amendment_status_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_status_code.permit_amendment_status_code'))
+    permit_amendment_type_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_type_code.permit_amendment_type_code'))
+
+    def json(self):
+        return {
+            'permit_amendment_id': str(self.permit_amendment_id),
+            'permit_amendment_guid': str(self.permit_amendment_guid),
+            'permit_id': self.permit_id,
```

Comment:
You touched on this in your tech demo, but I think I missed the reasoning. Why are we referencing the permit by id here instead of by guid? Is that the preferred pattern for relationships going forward? Or is it specific to permits/amendments?

Reply:
I thought the plan going forward was to use serial Id's where we could going forward along with the guid's. That way we can use the serial ID for the primary key and not propagate the guid's across the relationships to save space.

Reply:
Yep. We came across a ticket for that in the backlog in Sprint Planning 2 today. I wasn't aware of that, but now I am. Thanks!

Reply:
That's my understanding. As part of the DB changes Julian proposed auto-incrementing integers should be primary keys going forward as they are faster for queries and indexes, so I took this as an opportunity to add this to the permit table.